indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -493,23 +536,18 @@ public static List<TimelineObjectHolder<String, DataSegment>> getTimelineForInte
   {
     Preconditions.checkNotNull(interval);
 
-    // This call used to use the TaskActionClient, so for compatibility we use the same retry configuration
-    // as TaskActionClient.
     final RetryPolicy retryPolicy = retryPolicyFactory.makeRetryPolicy();
     Collection<DataSegment> usedSegments;
     while (true) {
       try {
-        usedSegments = coordinatorClient.fetchUsedSegmentsInDataSourceForIntervals(
-            dataSource,
-            Collections.singletonList(interval)
-        );
+        usedSegments = fetchUsedSegmentsForInterval(toolbox, coordinatorClient, dataSource, interval);
```

Comment:
Random nit: I'm not sure this really benefits from being a method?  It's not called from any other place, so it feels like it's just creating a method to create a method.  Might as well just in-line it.